libs/network/socketserver.c

```diff
@@ -317,47 +339,61 @@ bool socketServerConnectionsIdle(JsNetwork *net) {
     int error = 0;
 
     if (!closeConnectionNow) {
-      int num = netRecv(net, sckt, buf, net->chunkSize);
+      uint32_t host;
+      unsigned short port;
+      int num = netRecv(net, sckt, buf, (size_t)net->chunkSize, &host, &port);
       if (num<0) {
         // we probably disconnected so just get rid of this
         closeConnectionNow = true;
         error = num;
       } else {
-        // add it to our request string
         if (num>0) {
           JsVar *receiveData = jsvObjectGetChild(connection,HTTP_NAME_RECEIVE_DATA,0);
           JsVar *oldReceiveData = receiveData;
           if (!receiveData) receiveData = jsvNewFromEmptyString();
           if (receiveData) {
             jsvAppendStringBuf(receiveData, buf, (size_t)num);
-            bool hadHeaders = jsvGetBoolAndUnLock(jsvObjectGetChild(connection,HTTP_NAME_HAD_HEADERS,0));
-            if (!hadHeaders && httpParseHeaders(&receiveData, connection, true)) {
-              hadHeaders = true;
-              jsvObjectSetChildAndUnLock(connection, HTTP_NAME_HAD_HEADERS, jsvNewFromBool(hadHeaders));
-              JsVar *server = jsvObjectGetChild(connection,HTTP_NAME_SERVER_VAR,0);
-              JsVar *args[2] = { connection, socket };
-              jsiQueueObjectCallbacks(server, HTTP_NAME_ON_CONNECT, args, ((socketType&ST_TYPE_MASK)==ST_HTTP) ? 2 : 1);
-              jsvUnLock(server);
-            }
-            if (hadHeaders && !jsvIsEmptyString(receiveData)) {
-              // Keep track of how much we received (so we can close once we have it)
-              if ((socketType&ST_TYPE_MASK)==ST_HTTP) {
-                jsvObjectSetChildAndUnLock(connection, HTTP_NAME_RECEIVE_COUNT,
-                    jsvNewFromInteger(
-                      jsvGetIntegerAndUnLock(jsvObjectGetChild(connection, HTTP_NAME_RECEIVE_COUNT, JSV_INTEGER)) +
-                      jsvGetStringLength(receiveData)
-                    ));
+            if ((socketType&ST_TYPE_MASK)==ST_UDP) {
+              JsVar *address = jsvNewFromEmptyString(); // inet_ntoa replacement
+              JsVar *receiveInfo = jsvNewObject();
+              if (receiveInfo && address) { // could be out of memory
+                jsvAppendPrintf(address, "%d.%d.%d.%d", ((uint8_t*)&host)[0], ((uint8_t*)&host)[1], ((uint8_t*)&host)[2], ((uint8_t*)&host)[3]);
+                jsvObjectSetChildAndUnLock(receiveInfo, "address", address);
+                jsvObjectSetChildAndUnLock(receiveInfo, "port", jsvNewFromInteger(port));
+                jsvObjectSetChildAndUnLock(receiveInfo, "size", jsvNewFromInteger(num));
+                jswrap_dgram_messageCallback(connection, receiveData, receiveInfo);
```

Comment:
You could queue an event here instead I guess - that's what we seem to do elsewhere in the socket library. I'm not sure if it'd make much difference, but I guess the user could change the socket state under out feet this way round.